demo_nodes_cpp/src/parameters/list_parameters.cpp

```diff
@@ -27,7 +29,15 @@ int main(int argc, char ** argv)
   auto parameter_service = std::make_shared<rclcpp::parameter_service::ParameterService>(node);
 
   auto parameters_client = std::make_shared<rclcpp::parameter_client::SyncParametersClient>(node);
+  while (!parameters_client->wait_for_service(1s)) {
+    if (!rclcpp::ok()) {
+      std::cout << "Interrupted while waiting for the service. Exiting." << std::endl;
```

Comment:
This entire file uses streams. If changing to printf, all prints should be updated. This can be done either here or in a separate PR